local-dmabuf.c

```diff
@@ -55,6 +58,41 @@ struct dma_buf_sync {
 	uint64_t flags;
 };
 
+/* Return global DMA heap name from environment.
+ * Format:
+ *   LIBIIO_DMA_HEAP_PATH=heap_name
+ * Fallback to "system" if unset, empty, or not matching any accepted names.
+ */
+static const char *get_dma_heap_name(void)
+{
+	const char *env_value = getenv(LIBIIO_DMA_HEAP_ENV_VAR);
+	static const char *accepted_heaps[] = {
+		"system",
+		"default_cma_region",
+		"reserved",
+		"linux,cma",
+		"default-pool"
+	};
+	static const size_t num_accepted_heaps = sizeof(accepted_heaps) / sizeof(accepted_heaps[0]);
+	size_t i;
+
+	if (!env_value || !*env_value)
+		return "system";
+
+	for (i = 0; i < num_accepted_heaps; i++) {
+		if (!strcmp(env_value, accepted_heaps[i]))
+			return accepted_heaps[i];
+	}
+
+	/* Environment variable value is not in the accepted pool */
+	prm_warn(NULL, "LIBIIO_DMA_HEAP_PATH value '%s' is not recognized. "
+		"Accepted values are: \"system\", \"default_cma_region\", "
+		"\"reserved\", \"linux,cma\", \"default-pool\". "
+		"Defaulting to \"system\"\n", env_value);
+
+	return "system";
```

Comment:
Not sure if this should not be an error given that the user indeed made an error and given that this happens locally one cannot easily realize what's the issue.

That said, up to you